sw/device/silicon_creator/lib/drivers/keymgr_unittest.cc

```diff
@@ -105,49 +86,30 @@ TEST_F(KeymgrTest, StateAdvanceToCreator) {
           {KEYMGR_CONTROL_OPERATION_OFFSET,
            KEYMGR_CONTROL_OPERATION_VALUE_ADVANCE},
       });
-
-  EXPECT_EQ(
-      keymgr_state_advance_to_creator(&cfg_.binding_value, cfg_.max_key_ver),
-      kErrorOk);
-}
-
-TEST_F(KeymgrTest, StateAdvanceToCreatorInvalid) {
-  // Any state different than INIT is expected to fail.
-  ExpectStatusCheck(KEYMGR_OP_STATUS_STATUS_VALUE_DONE_SUCCESS,
-                    KEYMGR_WORKING_STATE_STATE_VALUE_RESET,
-                    /*err_code=*/0u);
-  EXPECT_EQ(
-      keymgr_state_advance_to_creator(&cfg_.binding_value, cfg_.max_key_ver),
-      kErrorKeymgrInternal);
-
-  // Any non-idle status is expected to fail.
-  ExpectStatusCheck(KEYMGR_OP_STATUS_STATUS_VALUE_DONE_ERROR,
-                    KEYMGR_WORKING_STATE_STATE_VALUE_INIT,
-                    /*err_code=*/0u);
-  EXPECT_EQ(
-      keymgr_state_advance_to_creator(&cfg_.binding_value, cfg_.max_key_ver),
-      kErrorKeymgrInternal);
+  keymgr_advance_state();
 }
 
-TEST_F(KeymgrTest, StateCreatorCheck) {
-  ExpectStatusCheck(KEYMGR_OP_STATUS_STATUS_VALUE_DONE_SUCCESS,
+TEST_F(KeymgrTest, CheckState) {
+  ExpectStatusCheck(KEYMGR_OP_STATUS_STATUS_VALUE_IDLE,
                     KEYMGR_WORKING_STATE_STATE_VALUE_CREATOR_ROOT_KEY,
                     /*err_code=*/0u);
-  EXPECT_EQ(keymgr_state_creator_check(), kErrorOk);
+  EXPECT_EQ(keymgr_check_state(kKeyMgrStateCreatorRootKey), kErrorOk);
 }
 
-TEST_F(KeymgrTest, StateCreatorCheckInvalidResponse) {
+TEST_F(KeymgrTest, CheckStateInvalidResponse) {
   // Any state different than CREATOR_ROOT is expected to fail.
-  ExpectStatusCheck(KEYMGR_OP_STATUS_STATUS_VALUE_DONE_SUCCESS,
+  ExpectStatusCheck(KEYMGR_OP_STATUS_STATUS_VALUE_IDLE,
                     KEYMGR_WORKING_STATE_STATE_VALUE_INVALID,
                     /*err_code=*/0u);
-  EXPECT_EQ(keymgr_state_creator_check(), kErrorKeymgrInternal);
+  EXPECT_EQ(keymgr_check_state(kKeyMgrStateCreatorRootKey),
+            kErrorKeymgrInternal);
 
   // Any non-idle status is expected to fail.
   ExpectStatusCheck(KEYMGR_OP_STATUS_STATUS_VALUE_DONE_ERROR,
                     KEYMGR_WORKING_STATE_STATE_VALUE_CREATOR_ROOT_KEY,
                     /*err_code=*/0u);
-  EXPECT_EQ(keymgr_state_creator_check(), kErrorKeymgrInternal);
+  EXPECT_EQ(keymgr_check_state(kKeyMgrStateCreatorRootKey),
+            kErrorKeymgrInternal);
 }
```

Comment:
should this test also cover `err_code != 0`?

Reply:
Done.